### ISSUE TITLE
Support tab completion / custom key handlers in QuickPick

### ISSUE THREAD

**Issue by Osmose** [microsoft/vscode#107265]
In https://github.com/jit-y/vscode-advanced-open-file, we create a QuickPick with a full file path that live-updates as you type with files matching the path. I'd like to add tab-completion so that, when you hit tab with the dialog open, it autocompletes the path if there's only one option visible, and if there are multiple matches, autocompletes up to the common prefix between them.

API-wise, it'd be nice to be able to register a keydown event handler or something for when the text input is focused so that we can read whether the user hit Tab and decide how to respond.

Currently, tab switches focus to the list of choices, which seems redundant to me since the choice list responds to input even without focus (arrow keys and enter still work). This might be an accessibility thing, though? 

Thank you!

**Comment by chrmarti** [comment]
Yes, Tab functionality is needed for accessibility. The editor area has the same issue though and is using a 'Tab moves focus' mode to make it accessible.

**Comment by stevenguh** [comment]
@Osmose I think this is indeed possible with the help of keybinding. See a change I did with a similar extension @ https://github.com/bodil/vscode-file-browser/commit/39ed84c1d018ca361b04c301455d58ba80bc95a9
